src/compiler/transformers/es6.ts

```diff
@@ -834,12 +897,38 @@ namespace ts {
             return block;
         }
 
-        function transformConstructorBodyWithSynthesizedSuper(node: ConstructorDeclaration) {
-            return visitNodes(node.body.statements, visitor, isStatement, 1);
+        function makeTransformerForConstructorBodyAtOffset(offset: number): (c: ConstructorDeclaration) => NodeArray<Statement> {
+            return constructor => visitNodes(constructor.body.statements, visitor, isStatement, /*start*/ offset);
         }
 
-        function transformConstructorBodyWithoutSynthesizedSuper(node: ConstructorDeclaration) {
-            return visitNodes(node.body.statements, visitor, isStatement, 0);
+        /**
+         * We want to try to avoid emitting a return statement in certain cases if a user already returned something.
+         * It would be pointless and generate dead code, so we'll try to make things a little bit prettier

```

Comment:
it should actually read as "pointless and would generate dead code", but that's redundant and unnecessary